### PR TITLE
Enable WDigest Windows Post Module

### PR DESCRIPTION
**Pull request by Meatballs1**
Doing pull request for a colleague...

This module simply enables the WDigest Credential provider for later versions of Windows. This allows us to 'mimikatz' users who login after this change.
## Verification
- [x] run creds_wdigest on and see no creds
- [x] Run this module then re-login
- [x] run creds_wdigest on and see creds

### CODE REVIEW

**Comment by zeroSteiner** [comment]
I think this should also have the ability to disable wdigest and be renamed to imply that it can both enable and disable the setting. Also supporting both would make testing just slightly easier.

**Comment by Meatballs1** [comment]
Rebased on master with enable/disable functionality.

**Comment by zeroSteiner** [comment]
Tested with a Windows 8.1 x64 system and it's working as intended.

```
meterpreter > run post/windows/manage/wdigest_caching 

[*] [2016.03.29-18:47:49] Running module against WINDOWS8VM
[*] [2016.03.29-18:47:49] Checking if the HKLM\SYSTEM\CurrentControlSet\Control\SecurityProviders\WDigest\UseLogonCredential DWORD exists...
[*] [2016.03.29-18:47:49] Creating UseLogonCredential DWORD value as 1...
[+] [2016.03.29-18:47:49] WDigest Security Provider enabled
meterpreter > run post/windows/manage/wdigest_caching 

[*] [2016.03.29-18:47:58] Running module against WINDOWS8VM
[*] [2016.03.29-18:47:58] Checking if the HKLM\SYSTEM\CurrentControlSet\Control\SecurityProviders\WDigest\UseLogonCredential DWORD exists...
[*] [2016.03.29-18:47:58] UseLogonCredential is set to 1
[+] [2016.03.29-18:47:58] Registry value is already set. WDigest Security Provider is enabled
meterpreter > 
```

**Comment by OJ** [comment]
Wow, was this really PR'd in September?! Sheesh man, we gotta get through these PRs quicker :/

File: modules/post/windows/manage/wdigest_caching.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
+require 'msf/core'
+require 'rex'
+require 'msf/core/auxiliary/report'
+
+class Metasploit3 < Msf::Post
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Registry
+  include Msf::Post::Windows::Accounts
+  include Msf::Auxiliary::Report
```
Comment (zeroSteiner) [inline modules/post/windows/manage/wdigest_caching.rb]: Out of all of these includes, it looks like only `Msf::Post::Windows::Registry` is required. Is there something that I'm missing? I don't see any references to functions provided by the other mixins.

File: modules/post/windows/manage/wdigest_caching.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  WDIGEST_REG_LOCATION = 'HKLM\\SYSTEM\\CurrentControlSet\\Control\\SecurityProviders\\WDigest'
+  USE_LOGON_CREDENTIAL = 'UseLogonCredential'
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'Windows Post Manage WDigest Credential Caching',
+      'Description'   => %q{
+          On Windows 8/2012 or higher, the Digest Security Provider (WDIGEST) is disabled by default. This module enables/disables
+          credential caching by adding/changing the value of the UseLogonCredential DWORD under WDIGEST provider's Registry key.
```
Comment (zeroSteiner) [inline modules/post/windows/manage/wdigest_caching.rb]: `DWORD under WDIGEST provider's` should be `DWORD under the WDIGEST provider's`

**Comment by zeroSteiner** [comment]
@OJ yes, I agree

**Comment by Meatballs1** [comment]
Error is about `MetasploitModule` naming convention. Is this a thing? :)

**Comment by wchen-r7** [comment]
@Meatballs1 yeah, instead of ```class Metasploit3 ```, we are using ```class MetasploitModule``` now.

**Comment by zeroSteiner** [comment]
Tested this one more time and everything looks good. Will land it in just a minute.

**Pull request merged**